### DISCUSSION TITLE
Any recommendations on providing common tests from a nuget package?

### DISCUSSION THREAD

**Discussion by forrestab** [xunit/xunit#2687] [Question] [answered]
We have several apis that all use the same authentication package. We are in the process of writing "integration" tests specific to the authentication package using `WebApplicationFactory<>`. What we would like to do is provide an authentication.testing package that the api's integration tests project will consume to provide those common integration tests for authentication.

Currently what we've done is moved the tests into the testing package and in the api's integration tests project there is a "dummy" class that inherits from those common test classes making them available to the api's test explorer.

```csharp
// Authentication.Testing package
class AuthenticationHandlerTests<TStartup> where TStartup : class
{
    // WebApiFactory<TStartup> is our customized WebApplicationFactory<T> class
    public AuthenticationHandlerTests(WebApiFactory<TStartup> factory)
    {
        // ...
    }

    // tests...
}

// Api.IntegrationTests project
class AuthenticationHandlerTests : 
    AuthenticationHandlerTests<Program>, IClassFixture<WebApiFactory<Program>>
{
    public AuthenticationHandlerTests(WebApiFactory<Program> factory)
        : base(factory)
    { }
}
```

Is this the right way or are we way off base? Any suggestions would be appreciated.

**Comment by bradwilson** [comment, marked as answer]
I've never heard of anybody doing this, but I don't see why not. The only thing to be wary of is the fact that people can't change or enhance the tests as easily as if you'd just provided them as source (you can overcome some of this with careful design of the test classes, depending on what your needs are).
